From 96f7edf8a21d247a7d170dbb2596f6f5069bee9a Mon Sep 17 00:00:00 2001
From: Sharaf Zaman <sharafzaz121@gmail.com>
Date: Fri, 29 May 2020 00:22:50 +0530
Subject: [PATCH 4/4] Andriod make URIs writable

If in future, file manager ever asks for readable permissions
only then we can consider having a dynamic permissions setup.

Change-Id: If25d6e3a3e3f6a4899e43a196f1e692f64265ac7
---
 src/plugins/platforms/android/androidcontentfileengine.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/plugins/platforms/android/androidcontentfileengine.cpp b/src/plugins/platforms/android/androidcontentfileengine.cpp
index 3e3bdc2592..6351b642ae 100644
--- a/src/plugins/platforms/android/androidcontentfileengine.cpp
+++ b/src/plugins/platforms/android/androidcontentfileengine.cpp
@@ -90,7 +90,7 @@ qint64 AndroidContentFileEngine::size() const
 
 AndroidContentFileEngine::FileFlags AndroidContentFileEngine::fileFlags(FileFlags type) const
 {
-    FileFlags commonFlags(ReadOwnerPerm|ReadUserPerm|ReadGroupPerm|ReadOtherPerm|ExistsFlag);
+    FileFlags commonFlags(ReadOwnerPerm|ReadUserPerm|ReadGroupPerm|ReadOtherPerm|ExistsFlag|WriteOwnerPerm|WriteUserPerm|WriteGroupPerm|WriteOtherPerm);
     FileFlags flags;
     const bool exists = QJNIObjectPrivate::callStaticMethod<jboolean>(
             "org/qtproject/qt5/android/QtNative", "checkFileExists",
-- 
2.26.2

